### ISSUE TITLE
Parameterized csvSource with body payload from reusable external JSON file

### ISSUE THREAD

**Issue by authorjapps** [authorjapps/zerocode#409]
From Slack:
@Dorian

Hello,
is it possible to use csv data in a reused son?

Because the ${0} like variables in the file aren't getting swapped with the values from the csv.

```JSON
...
"body": "${JSON.FILE:notification-api/parameterizable_body_pos_csv.json}" 
...
"parameterized": {
    "csvSource": [
      "Amelia, Loska, 1969-02-06, female, Im Grunde 082, Thörnich, 54340",
      "Pauline, Brand, 1967-01-29, female, Schöneberger Str. 71, Strande, 24229"
    ]
  }
```

This looks like anew feature request which needs to be implemented.

Please recheck the behavior before implementing it.

Current:
Inline CSV data in the `"parameterized"` block work fine

Desired:
CSV data from external source should work in similar fashion.

AC Details
===
From the perspective of the changes, the following is a better way. e.g.
Currently, it works this way.
```JSON
    "parameterized": {
        "csvSource":[
            "octocat,      The Octocat, San Francisco, 583231",
            "siddhagalaxy, Sidd,        UK,            33847730"
        ]
    }
```
What should be ideal is to source this from a CSV file e.g.
```JSON
    "parameterized": {
        "csvSource":"${CSV.FILE:my_data/params.csv}"
    }
```
where the `params.csv` content will be e.g.
```csv
user,name,city,userid
octocat,The Octocat,San Francisco,583231
siddhagalaxy,Sidd,UK,33847730
```

The 1st line is the header which can be ignored while reading.

**Comment by pradeeppg89** [comment]
I would like to help but  am totally new to this project. Can you please elaborate more on this?

**Comment by authorjapps** [comment]
@pradeeppg89 , thanks for your interest in contributing.
Current:
Inline CSV data in the `"parameterized"` block work fine

Desired:
CSV data from external source should work in similar fashion.

**Comment by authorjapps** [comment]
@pradeeppg89, are you still looking into this?

**Comment by Sauhardstark** [comment]
Hi @authorjapps 
I'm trying to understand the above issue and I understood that something like the below might be required.

````

"parameterized": {
    "csvFileSource": {
     test.json
    }
  }

````

Now the `csvFileSource` can also be a list. I'm more or less sure that this is wrong. Could you please correct my understanding here.

Also I'm a little unclear on the reused son part. Can we please have a chat regarding this on Slack

**Comment by authorjapps** [comment]
@Sauhardstark 
I think this is something I believe, should be reproduced as explained in the issue description.

I find your approach more simpler and elegant.
```
"parameterized": {
    "csvFileSource": "${JSON.FILE:notification-api/csv_data.json}"
  }
```

where `csv_data.json` is:
```
[
      "Amelia, Loska, 1969-02-06, female, Im Grunde 082, Thörnich, 54340",
      "Pauline, Brand, 1967-01-29, female, Schöneberger Str. 71, Strande, 24229"
]
```

Can you check if the params like "${0}" gets resolved automatically?

Depending on that we should proceed for the fix.

_This should be categorized under "Issues" rather than "Feature Requests"_

**Comment by ShubhamAnand2098** [comment]
Hi @authorjapps
Is this issue fixed? if not I would like to try solving this

**Comment by imprashant** [comment]
Hello @authorjapps 
I have raised a PR for this, can you please let me know if the PR is as per the expectation?
Also is it OK to assign this to me?

**Comment by authorjapps** [comment]
Hello @imprashant ,  Apologies for the delay as we were slightly busy in our day jobs.
We will take a look soon.

**Comment by arjunjhnzpost** [comment]
This would be a cool feature to have, really.

**Comment by manishvancouver** [comment]
Similar requirement has been requested in https://github.com/authorjapps/zerocode-hello-world/issues/29 as well.
Parameterization of external files will be important feature that could solve multiple challenges.

**Comment by authorjapps** [comment]
PR merged.

This feature is available in `1.3.36` (and higher version) when release.
